src/Main.js

```diff
@@ -1,23 +1,48 @@
 import beasts from './data.json';
 import HornedBeast from './HornedBeast';
 import { Component } from 'react';
+import FilterForm from './Form';
+import 'bootstrap/dist/css/bootstrap.min.css';
 
+const shuffleBeasts = shuffle(beasts.slice());
 
 class Main extends Component {
+
   constructor() {
-    super(constructor);
-    this.newBeasts = beasts.slice();
-    shuffle(this.newBeasts);
+    super();
+    this.state = {
+      beasts: shuffleBeasts,
+    };
+  }
+
+  handleChange = (e) => {
+    console.log(e);
+    const selection = e.target.value;
+    if (selection === '1'){
+      this.setState({beasts: shuffleBeasts.filter(beast => beast.horns === 1)});
+    }else if (selection === '2'){
+      this.setState({beasts: shuffleBeasts.filter(beast => beast.horns === 2)});
+    }else if (selection === '3'){
+      this.setState({beasts: shuffleBeasts.filter(beast => beast.horns === 3)});
+    }else if (selection === 'alot'){
+      this.setState({beasts: shuffleBeasts.filter(beast => beast.horns >= 4)});
+    }else{
+      this.setState({beasts: shuffleBeasts});
+    }
   }
+
```

Comment:
Hi there, nice work again on this lab!

I had almost forgotten to leave some feedback about this block and your `renderOnClickHandle()`. 

Multiple `setState` calls in the handler and this block may be something you want to take a closer look at. 

For example, the multiple setState calls in the handler can all be done in one pass/one call as setState accepts an object.

For this block I would think about how you could have rewritten this should you had more time. 

Perhaps you could have made a filter function and then used that in the `handleChange()`.

Just some food for thought for next time 👍